packages/runtime/lifecycle/src/api.ts

```diff
@@ -28,14 +38,177 @@ export type RuntimeLifecycle = {
+      *        "e68f1cea19752d1292b4be71b7f5d2b3219a15859c028f7454f66cdf",
+      *        ["role_A","role_C"]
+      *      )
+      * ```
+      *
+      * @see
+      * - {@link @marlowe.io/runtime-rest-client!contract.useMintedRoles}
+      * - {@link @marlowe.io/runtime-rest-client!contract.mintRole}
+      * - Open Roles Runtime Implementation : https://github.com/input-output-hk/marlowe-cardano/blob/main/marlowe-runtime/doc/open-roles.md
+      */
+  rolesConfiguration?: RolesConfiguration;
+
+  /**
+   * Marlowe Tags are stored as Metadata within the Transaction Metadata under the top-level Marlowe Reserved Key (`1564`).
+   * Tags allows to Query created Marlowe Contracts via {@link @marlowe.io/runtime-rest-client!index.RestClient#getContracts | Get contracts }
+   *
+   * <h4>Properties</h4>
+   *
+   * 1. They aren't limited size-wise like regular metadata fields are over Cardano.
+   * 2. Metadata can be associated under each tag
+   *
+   * @example
+   * ```ts
+   * const myTags : Tags = { "My Tag 1 That can be as long as I want": // Not limited to 64 bytes
+   *                            { a: 0
+   *                            , b : "Tag 1 content" // Limited to 64 bytes (Cardano Metadata constraint)
+   *                            },
+   *                         "MyTag2": { c: 0, d : "Tag 2 content"}};
+   * ```
+   */
   tags?: Tags;
+  /**
+   * Cardano Metadata about the contract creation.
+   * <h4>Properties</h4>
+   * <p>
+   * Metadata can be expressed as a JSON object with some restrictions:
+   *   - All top-level keys must be integers between 0 and 2^64 - 1.
+   *   - Each metadata value is tagged with its type.
+   *   - Strings must be at most 64 bytes when UTF-8 is encoded.
+   *   - Bytestrings are hex-encoded, with a maximum length of 64 bytes.
+   *
+   * Metadata aren't stored as JSON on the Cardano blockchain but are instead stored using a compact binary encoding (CBOR).
+   * The binary encoding of metadata values supports three simple types:
+   *    - Integers in the range `-(2^64 - 1)` to `2^64 - 1`
+   *    - Strings (UTF-8 encoded)
+   *    - Bytestrings
+   *    - And two compound types:
+   *        - Lists of metadata values
+   *        - Mappings from metadata values to metadata values
+   * </p>
+   * It is possible to transform any JSON object into this schema (See https://developers.cardano.org/docs/transaction-metadata )
+   * @see
+   * https://developers.cardano.org/docs/transaction-metadata
+   */
   metadata?: Metadata;
-  minUTxODeposit?: number;
-};
 
-export const minUTxODepositDefault: number = 3_000_000;
+  /**
+   * Minimum Lovelace value to add on the UTxO created (Representing the Marlowe Contract Created on the ledger).This value
+   * is computed automatically within the Runtime, so this parameter is only necessary if you need some custom adjustment.
+   *
+   * <h4>Justification</h4>
+   * <p>Creating a Marlowe Contracts over Cardano is about creating UTxO entries on the Ledger.
```

Comment:
"Creating a Marlowe Contracts"
should either be
"Creating Marlowe contracts"
or
"Creating a Marlowe contract"